runner.go

```diff
@@ -491,6 +515,63 @@ func runnerFromContainer(name string) (*runner, error) {
 	}, nil
 }
 
+// runOnOpen runs the image's `on_open` label in the container in the project directory.
+func (r *runner) runOnOpen(ctx context.Context, image string) error {
+	cli := dockerClient()
+	defer cli.Close()
+
+	// get project directory.
+	projectDir, err := r.projectDir(image)
+	if err != nil {
+		return err
+	}
+
+	// get on_open label from image
+	img, _, err := cli.ImageInspectWithRaw(context.Background(), image)
+	if err != nil {
+		return xerrors.Errorf("failed to inspect image: %w", err)
+	}
+	onOpenCmd, ok := img.Config.Labels[onOpenLabel]
+	if !ok {
+		// no on_open label, so we quit early.
+		return nil
+	}
+
+	cmd := []string{onOpenCmd}
+	return r.runInContainer(ctx, expandDir(projectDir), cmd, true)
+}
+
+// runInContainer runs a command in the container (optionally using /bin/sh -c) using exec (detached).
+func (r *runner) runInContainer(ctx context.Context, workDir string, cmd []string, useSh bool) error {
+	cli := dockerClient()
+	defer cli.Close()
+
+	if useSh {
+		cmd = append([]string{"/bin/sh", "-c"}, cmd...)
+	}
+
+	execID, err := cli.ContainerExecCreate(ctx, r.cntName, types.ExecConfig{
+		Cmd:        cmd,
+		Detach:     true,
+		WorkingDir: workDir,
+
+		// the following options don't attach it, but makes the script think it's running in a terminal.
+		Tty:          true,
+		AttachStdin:  true,
+		AttachStderr: true,
+		AttachStdout: true,
+	})
+	if err != nil {
+		return xerrors.Errorf("failed to create exec configuration: %v", err)
+	}
+
+	err = cli.ContainerExecStart(ctx, execID.ID, types.ExecStartCheck{Detach: true})
+	if err != nil {
+		return xerrors.Errorf("failed to start exec process: %v", err)
+	}
+	return nil
```

Comment:
The better way to do this is wrapping `docker exec -it` with the command to be executed. This lets us work with an `exec.Cmd` and prevents this function's signature from bloating.

Reply:
That should already be implemented somewhere in this codebase.

Reply:
Had to make a new function in `internal/dockutil/exec.go` as the only detached function didn't take a work directory, and from my tests this was required (`docker exec -it cdr_sail pwd` returns `/`).